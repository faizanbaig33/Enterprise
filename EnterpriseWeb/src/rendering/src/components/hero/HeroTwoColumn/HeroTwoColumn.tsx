import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { Headline } from 'src/helpers/Headline';
import { useTheme } from 'lib/context/ThemeContext';
import { HeroTwoColumnTheme } from './HeroTwoColumn.theme';
import { Component } from 'src/helpers/Component';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { Subheadline } from 'src/helpers/Subheadline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { ButtonGroup } from 'src/helpers/ButtonGroup';

export type HeroTwoColumnProps = Feature.EnterpriseWeb.Components.Hero.HeroTwoColumn;

const HeroTwoColumn = (props: HeroTwoColumnProps) => {
  const { themeName, themeData } = useTheme(HeroTwoColumnTheme);

  if (!props.fields) return null;

  return (
    <Component variant="lg" dataComponent="hero/herotwocolumn" gap="!gap-xxxs md:gap-s" {...props}>
      {themeName === 'rba' && (
        <div className="col-span-12 md:col-span-1">
          <span className="inline-block h-[3px] w-l bg-primary md:w-full"></span>
        </div>
      )}
      <div className={`col-span-12 ${themeName === 'rba' ? 'md:col-span-5' : 'md:col-span-6'}`}>
        <Headline useTag="h1" classes={themeData.classes.headlineClass} {...props} />
      </div>
      <div className="col-span-12 md:col-span-6">
        <Subheadline useTag="h2" classes={themeData.classes.subheadlineClass} {...props} />
        <BodyCopy classes={themeData.classes.bodyClass} {...props} />
        <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<HeroTwoColumnProps>(HeroTwoColumn);
