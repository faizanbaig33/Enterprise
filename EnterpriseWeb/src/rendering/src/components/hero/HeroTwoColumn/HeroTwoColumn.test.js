// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import HeroTwoColumn from './HeroTwoColumn';
import defaultData from './HeroTwoColumn.mock-data';

it('renders correctly', () => {
  const component = snapshot(HeroTwoColumn, { componentProps: defaultData });
  hasDataComponent(component, 'hero/herotwocolumn');
});
