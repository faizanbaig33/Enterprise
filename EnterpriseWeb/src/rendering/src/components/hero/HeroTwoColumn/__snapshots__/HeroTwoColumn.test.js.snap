// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="hero/herotwocolumn"
  id="HeroTwoColumn"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m !gap-xxxs md:gap-s"
    >
      <div
        class="col-span-12 md:col-span-6"
      >
        <div
          class="items-top font-sans  text-sm-l lg:text-l font-bold mb-s"
        >
          <h1>
            Hero heading goes here
          </h1>
        </div>
      </div>
      <div
        class="col-span-12 md:col-span-6"
      >
        <div
          class="font-sans text-sm-m lg:text-m font-medium mb-s"
        >
          <h2>
            Hero subheading goes here
          </h2>
        </div>
        <div
          class="text-body text-dark-gray font-regular mb-s"
        >
          <div
            class="body-copy"
          >
            <div>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc vulputate libero et velit interdum, ac aliquet odio mattis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
            </div>
          </div>
        </div>
        <div
          class="flex-col mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray mr-2 mb-m md:mb-0 mr-m"
            href="/"
            role="link"
            tabindex="0"
            title=""
          >
            Call to Action
            <span
              class="ml-xxs"
            />
          </a>
          <a
            class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray ml-xs md:ml-0 px-s md:px-0"
            href="/"
            role="link"
            tabindex="0"
            title=""
          >
            Call to Action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
